Outputing the function counter to XML.

giulianobelinassi opened the pull request:
Now, globalFunctions tag appears in the header of the xml output file.

Commit:
Outputing the function counter to XML.

Now, globalFunctions tag appears in the header of the xml output file.

rodrigosiqueira commented:
1. Change globalFunctions name to totalGlobalFunctions
2. When we execute kuniri with O1 flag it optimizes the final output to use a reduced version of the elements name. totalGlobalFunctions could be reduced to tgf and count to c when -O1 is passed to the CLI.
Thanks :)

lucaskanashiro committed:
Rename globalFunctions field in output and add short label

For now, globalFunctions = totalGlobalFunctions, and has 'tfg' as short label.

Signed-off-by: Lucas Kanashiro <kanashiro@debian.org>
Signed-off-by: Giuliano Belinassi <giuliano.belinassi@usp.br>

lucaskanashiro committed:
Fix xml output related tests

After changing globalFunctions to totalGlobalFunctions some tests broke.

Signed-off-by: Lucas Kanashiro <kanashiro@debian.org>
Signed-off-by: Giuliano Belinassi <giuliano.belinassi@usp.br>

rodrigosiqueira commented:
Almost done...
I just miss some tests similar the ones find in spec/language/ruby/global_test/. Add rspec tests with multiple variation of this parameter and other similar with tests in spec/language/ruby/global_test/

bsesso committed:
Tests for global functions counting

Signed-off-by: Bruno Sesso <bruno.sesso@usp.br>
Signed-off-by: Giuliano Belinassi <giuliano.belinassi@usp.br>

giulianobelinassi commented:
Done. Check spec/language/ruby/global_test/metrics

rodrigosiqueira commented:
Almost done, can you just rebase your branch with dev and fix the conflicts?

giulianobelinassi committed:
Merge branch 'dev' into functions_counter

Commit:
Fixed merge conflicts

Signed-off-by: giulianobelinassi <giuliano.belinassi@usp.br>
Signed-off-by: Lucas Kanashiro <kanashiro@debian.org>
Signed-off-by: Bruno Sesso <bsesso@gmail.com>

giulianobelinassi commented:
We do not know why the build is failing, but all tests succeed in my machine. @rodrigosiqueira Could you run it locally?
